src/Illuminate/Container/BoundMethod.php

```diff
@@ -105,18 +106,25 @@ protected static function normalizeMethod($callback)
      *
      * @param  \Illuminate\Container\Container  $container
      * @param  callable|string  $callback
-     * @param  array  $parameters
+     * @param  array $inputData
      * @return array
+     * @throws \ReflectionException
      */
-    protected static function getMethodDependencies($container, $callback, array $parameters = [])
+    protected static function getMethodDependencies($container, $callback, array $inputData = []) : array
     {
-        $dependencies = [];
+        $signature = static::getCallReflector($callback)->getParameters();
+
+        // In case the method has no explicit input parameters defined
+        // we will call it, with whatever input data available to us.
+        if (count($signature) === 0) {
```

Comment:
Is this to support `func_get_args()` in the called method? If so, I would like the comments to reflect this. Thanks

Reply:
sure